plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/PostgreUtils.java

```diff
@@ -120,21 +121,12 @@ public static <T extends PostgreAttribute> T getAttributeByNum(Collection<T> att
         return null;
     }
 
-    public static boolean isPGObject(Object object) {
-        if (object == null) {
-            return false;
-        }
-        String className = object.getClass().getName();
-        return className.equals(PostgreConstants.PG_OBJECT_CLASS) ||
-            className.equals(PostgreConstants.RS_OBJECT_CLASS) ||
-            className.equals(PostgreConstants.EDB_OBJECT_CLASS);
-    }
-
-    public static Object extractPGObjectValue(Object pgObject) {
+    @Nullable
+    public static Object extractPGObjectValue(@Nullable Object pgObject, @Nullable DBPDataSource dataSource) {
         if (pgObject == null) {
             return null;
         }
-        if (!isPGObject(pgObject)) {
+        if (!isPgObject(dataSource, pgObject)) {
             return pgObject;
         }
```

Comment:
I don't see a reason for creating a utility method here, especially when it checks for ins
       if (pgObject instanceof PostgreDataSource postgreDataSource && !postgreDataSource.getServerType().isPGObject(pgObject)) {
            return pgObject;
        }

Reply:
pgObject will never be a postgreDataSource, pgObject instanceof PostgreDataSource postgreDataSource is always falls